File: snippets/py/v0_5/concurrency/global.py
import inngest

inngest_client = inngest.Inngest(app_id="my-app")


# !snippet:start
@inngest_client.create_function(
    fn_id="func-a",
    trigger=inngest.TriggerEvent(event="ai/summary.requested"),
    concurrency=[
      inngest.Concurrency(
        scope="account",
        key='"openai"',
        limit=5
      )
    ]
)
async def func_a(ctx: inngest.Context):
    pass

@inngest_client.create_function(
    fn_id="func-b",
    trigger=inngest.TriggerEvent(event="ai/summary.requested"),
    concurrency=[
      inngest.Concurrency(
        scope="account",
        key='"openai"',
        limit=50
      )
    ]
)
async def func_b(ctx: inngest.Context):
    pass

File: snippets/go/v0_11/concurrency/global.go
package concurrency

import (
	"context"

	"github.com/inngest/inngest/pkg/enums"
	"github.com/inngest/inngestgo"
)

func GlobalSnippet() error {
	client, _ := inngestgo.NewClient(inngestgo.ClientOpts{AppID: "my-app"})

	// !snippet:start
	_, err := inngestgo.CreateFunction(
		client,
		inngestgo.FunctionOpts{
			Name: "func-a",
			Concurrency: []inngestgo.ConfigStepConcurrency{
				{
					Scope: enums.ConcurrencyScopeAccount,
					Key:   inngestgo.StrPtr(`"openai"`),
					Limit: 5,
				},
			},
		},
		inngestgo.EventTrigger("ai/summary.requested", nil),
		func(ctx context.Context, input inngestgo.Input[map[string]any]) (any, error) {
			return nil, nil
		},
	)
	if err != nil {
		return err
	}

	_, err = inngestgo.CreateFunction(
		client,
		inngestgo.FunctionOpts{
			Name: "func-b",
			Concurrency: []inngestgo.ConfigStepConcurrency{
				{
					Scope: enums.ConcurrencyScopeAccount,
					Key:   inngestgo.StrPtr(`"openai"`),
					Limit: 50,
				},
			},
		},
		inngestgo.EventTrigger("ai/summary.requested", nil),
		func(ctx context.Context, input inngestgo.Input[map[string]any]) (any, error) {
			return nil, nil
		},
	)
	if err != nil {
		return err
	}
	// !snippet:end

	return nil
}
